apps/blockchain/lib/blockchain/block.ex

```diff
@@ -940,4 +885,27 @@ defmodule Blockchain.Block do
       property_value -> property_value
     end
   end
+
+  defp add_metadata(block, trie, predefined_hash) do
+    block_rlp_size =
+      block
+      |> serialize
+      |> ExRLP.encode()
+      |> byte_size()
+
+    total_difficulty =
+      case get_block(block.header.parent_hash, trie) do
+        {:ok, block} ->
```

Comment:
would change this to `{:ok, parent_block}` because the `block` in this case is different than the `block` in the other case.